src/jsInterpreter.ts

```diff
@@ -0,0 +1,19 @@
+// @ts-ignore - no declared types for Neil Fraser's js-interpreter
+import Interpreter from "js-interpreter";
+
+// Add some basic functions to the interpreter
+const initFunc = function (interpreter: any, globalObject: any) {
+  const console = interpreter.nativeToPseudo({});
+  interpreter.setProperty(globalObject, "console", console);
+
+  const consoleLogWrapper = function (text: string) {
+    console.log(text);
+  };
+
+  interpreter.setProperty(
+    console,
+    "log",
+    interpreter.createNativeFunction(consoleLogWrapper),
```

Comment:
Does that not lead back to the security issue (since you can now do shit like console.log.prototype I guess) ?

Reply:
No, from my understanding JS Interpreter does not allow access to the prototype of a function passed in in this manner. This is the recommended method for passing in a function, so unless that breaks the sandbox, this should not be an issue.

Reply:
From the [JS-Interpreter documentation](https://neil.fraser.name/software/JS-Interpreter/docs.html):
> Additionally, API calls may be added to the interpreter's global object during creation. Here is the addition of alert() and a url variable:

    var myCode = 'alert(url);';
    var initFunc = function(interpreter, globalObject) {
      interpreter.setProperty(globalObject, 'url', String(location));

      var wrapper = function alert(text) {
        return window.alert(text);
      };
      interpreter.setProperty(globalObject, 'alert',
          interpreter.createNativeFunction(wrapper));
    };
    var myInterpreter = new Interpreter(myCode, initFunc);
  

> Alternatively, API calls may be added to an object:

    var myCode = 'robot.forwards(robot.fast);';
    var initFunc = function(interpreter, globalObject) {
      // Create 'robot' global object.
      var robot = interpreter.nativeToPseudo({});
      interpreter.setProperty(globalObject, 'robot', robot);

      // Define 'robot.fast' property.
      interpreter.setProperty(robot, 'fast', 99);

      // Define 'robot.forwards' function.
      var wrapper = function forwards(speed) {
        return realRobot.forwards(speed);
      };
      interpreter.setProperty(robot, 'forwards',
          interpreter.createNativeFunction(wrapper));
    };
    var myInterpreter = new Interpreter(myCode, initFunc);

This code essentially acts like the last example, but substituting `console` for `robot` and the `forwards` function for `log`.

Reply:
Right, since it's a wrapper you can't do JS shenanigans. Pretty neat